### PR TITLE
Allow variable length message in ed_verify

### PR DESCRIPTION
**Pull request by SwayStar123**
## Description
Change ed_verify to use a Bytes for the msg instead of b256 for a msg_hash
## Checklist

- [ ] I have linked to any relevant issues.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by JoshuaBatty — COMMENTED** [review]
Nice, can we also add a test case for the maximum allowed message length.

**Review by IGI-111 — APPROVED** [review]

**Comment by Voxelot** [comment]
I'd also like to note that we have a weird edge case if the length is set to 0, we default to 32 for backwards compatibility reasons. It might be best to just reject `len: 0` at the std-lib level to avoid any surprises.

**Comment by SwayStar123** [comment]
The tests actually fail and i cant figure out why @IGI-111 @Voxelot 
Am i correct in understanding the len parameter is for how long the message is before hashing?

**Review by alfiedotwtf — APPROVED** [review]

**Review by bitzoic — APPROVED** [review]

**Review by K1-R1 — APPROVED** [review]

**Review by IGI-111 — APPROVED** [review]

File: test/src/in_language_tests/test_programs/ecr_inline_tests/src/main.sw [hunk 1]
```diff
@@ -106,28 +107,103 @@ fn ecr_ec_recover_address_r1() {
...
+// Calculated with ed25519-dalek in a rust program
 
-    let hi_1 = 0xf38cef9361894be6c6e0eddec28a663d099d7ddff17c8077a1447d7ecb4e6545;
-    let lo_1 = 0xf5084560039486d3462dd65a40c80a74709b2f06d450ffc5dc00345c6b2cdd00;
-    let signature_1: B512 = B512::from((hi_1, lo_1));
-    // A verified public key with signature 
-    let verified_1 = ed_verify(pub_key_1, signature_1, msg_hash_1);
-    assert(verified_1.is_ok());
-    assert(verified_1.unwrap());
+// use ed25519_dalek::ed25519::signature::Signer;
```
Comment (alfiedotwtf) [inline test/src/in_language_tests/test_programs/ecr_inline_tests/src/main.sw:112]: Just checking: did you mean to leave this old test in a comment?
